modules/saml/lib/IdP/SAML2.php

```diff
@@ -218,6 +232,41 @@ private static function getAssertionConsumerService(array $supportedBindings, Si
 		return $spMetadata->getDefaultEndpoint('AssertionConsumerService', $supportedBindings);
 	}
 
+ 	/**
+	 * Check a SOAP AuthnRequest.
+	 *
+	 * @param SimpleSAML_Configuration $idpMetadata  The metadata for the IdP.
+	 * @param SimpleSAML_Configuration $spMetadata  The metadata for the SP.
+	 * @param array &$state  The state array.
+	 */
+	private static function processSOAPAuthnRequest(SimpleSAML_Configuration $idpMetadata, SimpleSAML_Configuration $spMetadata, SAML2_AuthnRequest $request, array &$state) {
+
+		$GLOBALS['SimpleSAML.debugDisableRedirect'] = TRUE;
```

Comment:
I understand this is a mechanism to make the implementation of the ECP client easier, so that it doesn't need to follow HTTP redirections. I don't know if the spec mandates it, but I think it's a good idea anyway.

The problem though is the use of globals. That's bad practice and I don't think we should accept it like that. The state array could be used instead to store a flag with the same purpose. However, `SimpleSAML_Utils_HTTP::redirect()`, where this variable should be checked instead of `SimpleSAML_Utilities::_doRedirect()`, does not receive the state array as a parameter, nor it should, IMHO, as its function should be pretty clear and simple: perform an HTTP redirection.

Is it possible to check the places that might be calling for a redirection, and check the state array there, avoiding the redirection if the flag is set?